libs/core/src/lib/token/token.component.ts

```diff
@@ -18,42 +16,44 @@ import {
     selector: 'fd-token',
     templateUrl: './token.component.html',
     styleUrls: ['./token.component.scss'],
-    host: {
-        class: 'fd-token',
-        '[class.fd-token__disabled]': 'disabled',
-        'role': 'button'
-    },
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class TokenComponent {
 
-    /** @hidden */
-    @ViewChild('contentContainer', { static: false })
-    contentContainer: ElementRef;
-
     /** Whether the token is disabled. */
     @Input()
     disabled: boolean = false;
 
+    /** Whether the token is compact. */
+    @Input()
+    compact: boolean = false;
+
+    /** Whether the token is selected. */
+    @Input()
+    selected: boolean = false;
+
+    /** Whether the token is read-only. */
+    @Input()
+    readOnly: boolean = false;
+
     /** Emitted when the *x* icon is clicked. Specifically, any pseudo-element. */
     @Output()
     readonly onCloseClick: EventEmitter<void> = new EventEmitter<void>();
 
     /** @hidden */
-    constructor(private elRef: ElementRef) {
-    }
+    elementRef: ElementRef;
 
     /** @hidden */
-    @HostListener('click', ['$event'])
     clickHandler(event): void {
-        if (this.contentContainer && !this.disabled) {
-            if (this.elRef.nativeElement.contains(event.target) && !this.contentContainer.nativeElement.contains(event.target)) {
-                this.onCloseClick.emit();
-            }
+        event.stopPropagation();
+        if (!this.disabled) {
+            this.onCloseClick.emit(event);
         }
     }
 
-}
-
+    constructor(elementRef: ElementRef) {
```

Comment:
The same situation as above, maybe consider adding `constructor(public elementRef: ElementRef) `